boards/waspmote-pro/Kconfig

```diff
@@ -11,11 +11,16 @@ config BOARD_WASPMOTE_PRO
     bool
     default y
     select CPU_MODEL_ATMEGA1281
+    select HAS_ARDUINO
     select HAS_PERIPH_ADC
     select HAS_PERIPH_I2C
     select HAS_PERIPH_SPI
     select HAS_PERIPH_TIMER
     select HAS_PERIPH_UART
-    select HAS_ARDUINO
+    select HAS_SDCARD_SPI
+
+    select MODULE_FATFS_VFS if MODULE_VFS_DEFAULT
+    select MODULE_MTD if MODULE_VFS_DEFAULT
+    select MODULE_SDCARD_SPI if MODULE_MTD
```

Comment:
I don't get it - we have

```
config HAVE_MTD_SDCARD
    bool 
    imply MODULE_MTD_SDCARD if MODULE_MTD
```

so this should not be needed.

And still CI complains about missing `mtd_sdcard` with KConfig

Reply:
The mtd is not really modelled according to the design patterns we are settling on and should be updated but that would be out of scope of this PR.  The `boards/waveshare-nrf52840-eval-kit/Kconfig` has a solution it seems.
This should solve it:

```diff

diff --git a/boards/waspmote-pro/Kconfig b/boards/waspmote-pro/Kconfig
index bdcaff16c5..a260a6501e 100644
--- a/boards/waspmote-pro/Kconfig
+++ b/boards/waspmote-pro/Kconfig
@@ -19,6 +19,9 @@ config BOARD_WASPMOTE_PRO
     select HAS_PERIPH_UART
     select HAS_SDCARD_SPI
 
+    select HAVE_MTD_SDCARD
+    select HAVE_SDCARD_SPI
+
     select MODULE_FATFS_VFS if MODULE_VFS_DEFAULT
     select MODULE_MTD if MODULE_VFS_DEFAULT
     select MODULE_SDCARD_SPI if MODULE_MTD
```